pkg/bootstrap/config.go

```diff
@@ -703,6 +703,12 @@ func GetNodeMetaData(options MetadataOptions) (*model.Node, error) {
 		untypedMeta[security.CredentialMetaDataName] = "true"
 	}
 
+	if !meta.MetadataDiscovery {
```

Comment:
nit: would slightly prefer to have it "if its **unset** default to meta.EnableHBONE". Not sure it matters much in practice, but its not really a feature flag if you cannot turn it off

Reply:
Agree it would be slightly better, but not sure how we can do that with a bool env var at the pilot-agent level

Reply:
The env flags have a Get() which is used now but also a Lookup() which retains that info. We use it in some other places (like `registerAmbient`)

Reply:
Thanks for the tip - was able to use that to convert all of the intermediate struct fields to `*bool`